File: src/src/junitjava/testcases/CalculationTest.java
package junitjava.testcases;

import junitjava.logic.Calculation;
import org.junit.*;

public class CalculationTest {

    @BeforeClass
    public static void setUpBeforeClass() throws Exception{
        System.out.println("First method to be invoked");
    }
    @Before
    public void setUp() throws Exception{
        System.out.println("Before testFindMax method");
    }
    @Test
    public void testFindMax()
    {
        System.out.println("Welcome");
        Assert.assertEquals(7, Calculation.findMax(new int[]{4,5,6,7,1,2,4,3}));
        Assert.assertEquals(0, Calculation.findMax(new int[]{-4,-5,-6,-7,-1,-2,-4,-3}));
    }
    @After
    public void setClose() throws Exception{
        System.out.println("After testFind Max method");
    }

    @Test(timeout = 1000)
    public void testFindCube(){
        Assert.assertEquals(27, Calculation.findCube(3));
    }

    @Test
    public void testReverseString(){
        Assert.assertEquals("marap", Calculation.reverseString("param"));
    }

    @AfterClass
    public static void setUpAfterClass()
    {
        System.out.println("From after class Final method");
    }
}

File: src/src/junitkotlin/testcases/CalculationTest.kt
package junitkotlin.testcases

import junitkotlin.logic.Calculation
import org.junit.*

class CalculationTest {

    companion object{
        @JvmStatic
        @BeforeClass
        fun setUpBeforeClass(){
            println("Before Class method invoked only once.")
        }

        @JvmStatic
        @AfterClass
        fun setUpAfterClass(){
            println("After class method invoked after all test cases.")
        }
    }

    @Before
    fun setUpBefore(){
        println("Before Method")
    }

    @Test
    fun testFindMax(){
        val arr = intArrayOf(1,2,3,4,5,6,7)
        println("FindMax")
        Assert.assertEquals(7, Calculation.findMax(arr))
    }

    @Test
    fun findCube(){
        println("FindCube")
        Assert.assertEquals(27, Calculation.findCube(3))
    }

    @After
    fun setUpAfter(){
        println("After Method")
    }

}